src/sqlfluff/rules/L016.py

```diff
@@ -481,28 +486,37 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         self.ignore_comment_lines: bool
         self.ignore_comment_clauses: bool
 
-        if not context.memory:
-            memory: dict = {"comment_clauses": set()}
-        else:
-            memory = context.memory
-        if context.segment.is_type("newline"):
-            # iterate to buffer the whole line up to this point
-            this_line = self._gen_line_so_far(context.raw_stack)
-        else:
-            if self.ignore_comment_clauses and context.segment.is_type(
-                "comment_clause", "comment_equals_clause"
-            ):
-                comment_segment = context.functional.segment.children().first(
-                    sp.is_type("quoted_literal")
-                )
-                if comment_segment:
-                    memory["comment_clauses"].add(comment_segment.get())
+        assert context.segment.is_type("newline")
 
-            # Otherwise we're all good
-            return LintResult(memory=memory)
+        # iterate to buffer the whole line up to this point
+        this_line = self._gen_line_so_far(context.raw_stack)
+
+        # Do any literals on this line belong to a comment?
+        literals_in_comments: Sequence[BaseSegment] = []
+        if self.ignore_comment_clauses:
+            quoted_literals = Segments(*this_line).select(
+                select_if=sp.is_type("quoted_literal")
+            )
+            if quoted_literals:
+                self.logger.debug(
+                    "Comment Checking: Found quoted literals: %s", quoted_literals
+                )
+                root = context.parent_stack[0]
+
+                def is_in_comment(seg):
+                    path = root.path_to(seg)
+                    parent = path[-2]
```

Comment:
Are we guaranteed `path[-2]` won't cause an `IndexError`? A comment here about the structure we expect (i.e. why `-2`) would be helpful.

Reply:
Will add a docstring to explain and a clause for if `path` is too short.